starter-code/js/index.js

```diff
@@ -1,11 +1,52 @@
-var $cart = document.querySelector('#cart tbody');
-var $calc = document.getElementById('calc');
+var $cart = document.querySelector("#cart tbody");
+var $calc = document.getElementById("calc");
+var $create = document.getElementById("create");
 
-function updateSubtot($product) {
+var $delete = document
+  .querySelectorAll(".rm > button")
+  .forEach(function(button) {
+    button.onclick = deleteProduct;
+  });
+
+function updateSubtot(element) {
   // Iteration 1.1
+  let subTotal;
+  let cost = element.querySelector(".pu > span").innerText;
+  let quantity = element.querySelector(".qty > label > input").value;
+  let subtotDiv = element.querySelector(" td.subtot > span");
+
+  subtotDiv.innerHTML = cost * quantity;
+  return cost * quantity;
 }
 
 function calcAll() {
+  let totalValue = 0;
   // Iteration 1.2
+  document.querySelectorAll("#cart > tbody > tr").forEach(element => {
+    totalValue += updateSubtot(element);
+  });
+
+  document.querySelector("body > h2 > span").innerHTML = totalValue;
+}
+
+function deleteProduct(event) {
+  let currentEventTarget = event.currentTarget; //the delete button you clıcked on
+
+  if (currentEventTarget.parentNode) {
+    currentEventTarget.parentNode.parentNode.parentNode.removeChild(
+      currentEventTarget.parentNode.parentNode
+    );
+  }
 }
-$calc.onclick = calcAll;
\ No newline at end of file
+
+function addRow() {
```

Comment:
This seems to be half done. One way to solve this challenge is like this:

let oldEle = document.querySelector("#cart > tbody > tr:nth-child(1)") 
const newEle = oldEle.cloneNode(true)  // These two lines just copies the format of the shopping cart, so i don't have //to remake it

function createNewProduct() {
  let createDiv = document.querySelector("#cart > tfoot > tr") // This is the "create new product div"
  let name = createDiv.querySelector("td:nth-child(1) > input").value // This is the name input for the new product
  let priceUnit = createDiv.querySelector("td:nth-child(2) > input").value // This is the price per unit input for the new //product
  let newTrElement = newEle.cloneNode(true)  // this create a copy of the <tr> product element

  newTrElement.querySelector("td.name > span").innerHTML = name // sets the name of the new product
  newTrElement.querySelector("td.pu > span").innerHTML = priceUnit // sets the price per unit of the new product
  addDeleteOnClick(newTrElement) // adds the onclick for the delete button (if i want to remove the product again)

  document.querySelector("#cart > tbody").appendChild(newTrElement) // adds the element to the carts tbody
}